File: python/str-gradient.py
import sys
import time

# Create source string.
const_text = ""
# Check for command-line arguments.
if len(sys.argv) > 1:
    for i in range(1, len(sys.argv)):
        const_text += sys.argv[i] + " "
else:
    # Otherwise, just ask the user for an input string.
    const_text = input("Enter a string: ")

# Fix user string...
if not len(const_text):
    const_text = "Hello World!"
if len(const_text) < 2:
    print("Error: Input string must be at least 2 characters.")
    sys.exit()

# Display string as a gradient/wave.
string = const_text
while True:
    while len(string) > 1:
        string = string[0:len(string) - 1]
        time.sleep(0.05)
        print(string)
    while len(string) < len(const_text):
        string = const_text[0:len(string) + 1]
        time.sleep(0.05)
        print(string)

File: cplusplus/str-gradient/str-gradient.cpp
#include <cstdint>
#include <iostream>
#include <string>
#include <unistd.h>

int main(int argc, char* argv[]) {
    // Create a source string.
    std::string src_string;
    // Check for command-line arguments.
    if (argc > 1) {
        for (int i = 1; i < argc; i++) {
            src_string += argv[i];
            src_string += ' ';
        }
    } else {
        // Otherwise, just ask the user for an input string.
        std::cout << "Enter a String: ";
        std::string new_input;
        while(std::cin >> new_input) {
            src_string += new_input + ' ';
        }
    }

    // Display string as a gradient/wave.
    std::string print_string = src_string;
    while (1) {
        while (print_string.length() > 1) {
            print_string.resize(print_string.length() - 1, ' ');
            std::cout << print_string << '\n';
            usleep(50000);
        }
        while (print_string.length() < src_string.length()) {
            int prev_len = print_string.length();
            print_string = src_string;
            print_string.resize(prev_len + 1, ' ');
            std::cout << print_string << '\n';
            usleep(50000);
        }
    }
}
